### PR TITLE
Separate Rendezvous message handling from BLE session

### PR DESCRIPTION
**Pull request by kpschoedel**
With this change, BLE handling remains in RendezvousSession and
message processing independent of BLE moves to RendervousMessageHandler.

### CODE REVIEW

File: examples/wifi-echo/server/esp32/main/RendezvousMessageHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+extern CHIP_ERROR SetWiFiStationProvisioning(char * ssid, char * key);
+
+CHIP_ERROR RendezvousMessageHandler::HandleMessageReceived(System::PacketBuffer * buffer)
+{
+    CHIP_ERROR err         = CHIP_NO_ERROR;
+    const size_t bufferLen = buffer->DataLength();
+    char * key             = nullptr;
+    char * ssid            = nullptr;
+    char msg[bufferLen];
```
Comment (kedars) [inline examples/wifi-echo/server/esp32/main/RendezvousMessageHandler.cpp:33]: This makes the stack utilisation dependent on the received DataLength. Is the data length known to be capped to a threshold? We could be more deterministic by having the high threshold value in here?

Comment (kpschoedel) [reply examples/wifi-echo/server/esp32/main/RendezvousMessageHandler.cpp:33]: Good catch; I'm surprised we don't use -Wvla.

Comment (bzbarsky-apple) [reply examples/wifi-echo/server/esp32/main/RendezvousMessageHandler.cpp]: Could you please file a followup issue to use `-Wvla`?

**Comment by bzbarsky-apple** [comment]
@pan-apple Could you take a look, please?

**Review by pan-apple — APPROVED** [review]

**Comment by pan-apple** [comment]
tag: @vivien-apple

**Review by bzbarsky-apple — APPROVED** [review]

**Review by kedars — APPROVED** [review]

File: examples/wifi-echo/server/esp32/main/RendezvousMessageHandler.cpp (hunk 1 above)
Comment (kpschoedel) [reply examples/wifi-echo/server/esp32/main/RendezvousMessageHandler.cpp]: Filed https://github.com/project-chip/connectedhomeip/issues/2427 proposing `-Wvla`.

(For this PR, https://github.com/project-chip/connectedhomeip/pull/2418/commits/d9dc684ccba267ceca23f532a079fc77a8d170ce removes the VLA.)

**Review by mspang — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@saurabhst @BroderickCarlin @andy31415 could you take a look?

**Review by rwalker-apple — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**